docs/preview/03-Features/secrets/general.md

```diff
@@ -16,14 +16,41 @@ string secretVersion = secret.Version;
 DateTimeOffset? expirationDate = secret.Expires;
 ```
 
-# Raw secrets
+## Raw secrets
 In some scenarios you'd like to just get the secret value directly without any metadata.
 This is possible by calling the `...Raw...` variants on the `ISecretProvider` implementations.
 
 ```csharp
 string secretValue = await secretProvider.GetRawSecretAsync("EventGrid-AuthKey");
 ```
 
+## Synchronous secrets
+In some scenarios you'd like to retrieve secrets synchronously. A common situation is when you want to register a dependent service into the dependency container that requires a secret, but since such a container only registers instances or synchronous functions, there is no easy way to retrieve a secret asynchronously.
+
+Almost all built-in secret providers we provide support synchronous secret retrieval, only the [HashiCorp Vault secret provider](../secret-store/provider/hashicorp-vault.md) does not support this.
+
+Retrieving synchronous secrets can be done via either using the `ISyncSecretProvider` alternative interface, or by calling the `GetSecret` extensions on an `ISecretProvider` implementation.
+
+```csharp
+var services = new ServiceCollection();
+services.AddSingleton(serviceProvider =>
+{
+    // #1 injecting the `ISyncSecretProvider`:
+    var syncSecretProvider = serviceProvider.GetRequiredService<ISyncSecretProvider>();
+    Secret secret = synSecretProvider.GetSecret("EventGrid-AuthKey");
+    string secretValue = syncSecretProvider.GetRawSecret("EventGrid-AuthKey");
+
+    // #2 calling `GetSecret` or `GetRawSecret` extension on `ISecretProvider`:
+    var secretProvider = serviceProvider.GetRequiredService<ISecretProvider>();
+    Secret secret = secretProvider.GetSecret("EventGrid-AuthKey");
+    string secretProvider = secretProvider.GetRawSecret("EventGrid-AuthKey");
+
+    return new MyDependentService(secretValue);
+});
+```
+
+⚠ Make sure that you only call the `GetSecret` and `GetRawSecret` extension on `ISecretProvider` implementations that also implement the `ISyncSecretProvider` interface. The [Arcus secret store](../secret-store/index.md) automatically makes sure that you can use this extension on any injected `ISecretProvider` but when no secret provider is registered that supports synchronous secret retrieval, an `SecretNotFoundException` will be thrown nonetheless.
+
```

Comment:
Shouldn't we throw a `NotSupportedException` in that case.  I think it makes it more clear in that case of what is happening.   Throwing a  `SecretNotFoundException` might just confuse the user, as he's sure that there exists a secret with that specific key.

Reply:
There is an `InvalidOperationException` as an inner exception of the `SecretNotFoundException` to explain why the secret was not found. Wouldn't that be enough?

Reply:
Idk if I would throw the `SecretNotFoundException` in this case, as it might be confusing.  It's actually not really the reason of the problem.

Reply:
Ok, let's throw a `NotSupportedException` for this. 👍